src/Functions/FunctionStartsEndsWith.h

```diff
@@ -113,8 +108,8 @@ class FunctionStartsEndsWith : public IFunction
 
     DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
     {
-        if (!is_utf8 && isStringOrFixedString(arguments[0]) && isStringOrFixedString(arguments[1])
-            || isString(arguments[0]) && isString(arguments[1]))
+        if ((!is_utf8 && isStringOrFixedString(arguments[0]) && isStringOrFixedString(arguments[1]))
+            || (isString(arguments[0]) && isString(arguments[1])))
```

Comment:
This appears to be a fix for some incorrect behavior, I think. But it's okay not to mention it if nobody hit it.